src/Playwright/Core/Shared/RunSettingsParser.cs

```diff
@@ -0,0 +1,176 @@
+using System.Runtime.CompilerServices;
+using System.Runtime.Serialization;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+
+[assembly: InternalsVisibleToAttribute("Microsoft.Playwright.MSTest, PublicKey=0024000004800000940000000602000000240000525341310004000001000100059a04ca5ca77c9b4eb2addd1afe3f8464b20ee6aefe73b8c23c0e6ca278d1a378b33382e7e18d4aa8300dd22d81f146e528d88368f73a288e5b8157da9710fe6f9fa9911fb786193f983408c5ebae0b1ba5d1d00111af2816f5db55871db03d7536f4a7a6c5152d630c1e1886b1a0fb68ba5e7f64a7f24ac372090889be2ffb")]
+[assembly: InternalsVisibleToAttribute("Microsoft.Playwright.NUnit, PublicKey=0024000004800000940000000602000000240000525341310004000001000100059a04ca5ca77c9b4eb2addd1afe3f8464b20ee6aefe73b8c23c0e6ca278d1a378b33382e7e18d4aa8300dd22d81f146e528d88368f73a288e5b8157da9710fe6f9fa9911fb786193f983408c5ebae0b1ba5d1d00111af2816f5db55871db03d7536f4a7a6c5152d630c1e1886b1a0fb68ba5e7f64a7f24ac372090889be2ffb")]
+
+namespace Microsoft.Playwright.Core.Shared
+{
+    internal class RunSettingsParser
+    {
+        private readonly IDictionary<string, string> _settings;
+        public BrowserTypeLaunchOptions LaunchOptions;
+        public string BrowserName;
+
+        internal RunSettingsParser(IDictionary<string, string> settings)
+        {
+            _settings = settings;
+            LaunchOptions = ParseTestParameters<BrowserTypeLaunchOptions>(settings);
+            BrowserName = DetermineBrowserType();
+        }
+
+        private static T ParseTestParameters<T>(IDictionary<string, string> parameters)
+        where T : class, new()
+        {
+            var options = new T();
+            foreach (var parameter in parameters)
+            {
+                if (!string.IsNullOrEmpty(parameter.Value))
+                {
+                    ApplyParameter(parameter.Key, parameter.Value, options);
+                }
+            }
+            return options;
+        }
+
+        private static void ApplyParameter(string key, string value, object options)
+        {
+            // Try to match it by the exact name of the property.
+            var property = options.GetType().GetProperty(key);
+            if (property == null)
+            {
+                // If not found, we try to fallback to the JsonPropertyNameAttribute
+                property = options.GetType().GetProperties().Where(p => p.GetCustomAttributes(typeof(JsonPropertyNameAttribute), false).Where(a => ((JsonPropertyNameAttribute)a).Name == key).Any()).FirstOrDefault();
+                if (property == null)
+                {
+                    return;
+                }
+            }
+            var type = Nullable.GetUnderlyingType(property.PropertyType) ?? property.PropertyType;
+            switch (type)
+            {
+                case Type t when t == typeof(string):
+                    property.SetValue(options, value);
+                    break;
+                case Type t when type == typeof(bool):
```

Comment:
`Type t when t ==` for consistency with other branches? I'm not actually sure what `Type t` would yield in this particular expression, is this case covered with a test?

Reply:
will follow up with better tests.